AlfrescoApp/View Controllers/Text File View Controller/TextFileViewController.m

```diff
@@ -196,20 +196,54 @@ - (void)cancelButtonPressed:(id)sender
     
     if (self.textView.text.length > 0)
     {
-        NSString *alertTitleKey = self.editingDocument ? @"document.edit.button.discard" : @"createtextfile.dismiss.confirmation.title";
-        NSString *alertMessageKey = self.editingDocument ? @"document.edit.dismiss.confirmation.message" : @"createtextfile.dismiss.confirmation.message";
+        BOOL shouldShowAlertView = false;
         
-        UIAlertView *confirmationAlert = [[UIAlertView alloc] initWithTitle:NSLocalizedString(alertTitleKey, @"Discard Title")
-                                                                    message:NSLocalizedString(alertMessageKey, @"Discard Message")
-                                                                   delegate:self
-                                                          cancelButtonTitle:NSLocalizedString(@"Yes", @"Yes")
-                                                          otherButtonTitles:NSLocalizedString(@"No", @"No"), nil];
-        [confirmationAlert showWithCompletionBlock:^(NSUInteger buttonIndex, BOOL isCancelButton) {
-            if (isCancelButton)
+        //we check to see if we are in editing mode
+        if((self.editingDocument) && (self.documentContentPath))
+        {
+            NSError *error = nil;
+            NSString *fileContent = [[NSString alloc] initWithContentsOfFile:self.documentContentPath usedEncoding:NULL error:&error];
+            if(error == nil)
             {
-                dismissController();
+                if(self.textView.text.length != fileContent.length)
+                {
+                    //user has edited the file
+                    shouldShowAlertView = true;
```

Comment:
Minor: We use `YES` and `NO` instead of `true` and `false`

Also this could be greatly simplified to
```objectivec
shouldShowAlertView = !(self.textView.text.length == fileContent.length);
```